packages/pagination/__tests__/pagination.spec.ts

```diff
@@ -1,407 +1,377 @@
+      expect(wrapper.find('.slot-test').exists()).toBe(true)
     })
-    expect(wrapper.find('li.number.active').text()).toEqual('3')
-  })
 
-  test('currentPage: NaN', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        pageSize: 20,
-        total: 200,
-        currentPage: NaN,
-      },
+    test('test small layout', () => {
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              total: 100,
+              small: true,
+            })
+          }
+        },
+      })
+      expect(wrapper.vm.$el.classList.contains('el-pagination--small')).toBe(true)
     })
-    expect(wrapper.find('li.number.active').text()).toEqual('1')
-    expect(wrapper.vm.$el.querySelectorAll('li.number').length).toBe(7)
-  })
 
-  test('layout is empty', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        layout: '',
-      },
+    test('test with background', async () => {
+      const withBackground = ref(true)
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              total: 100,
+              background: withBackground.value,
+            })
+          }
+        },
+      })
+      expect(wrapper.find('.is-background').exists()).toBe(true)
+      withBackground.value = false
+      await nextTick()
+      expect(wrapper.find('.is-background').exists()).toBe(false)
     })
-    expect(wrapper.vm.$el.textContent).toEqual('')
-  })
-})
 
-describe('click pager', () => {
-  test('click ul', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        total: 1000,
-      },
+    test('test hide-on-single-page prop', async () => {
+      const hideOnSinglePage = ref(false)
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              total: 10, // deivded by default page-size(10), there will be only one page
+              hideOnSinglePage: hideOnSinglePage.value,
+            })
+          }
+        },
+      })
+      expect(wrapper.find('.el-pagination').exists()).toBe(true)
+      hideOnSinglePage.value = true
+      await nextTick()
+      expect(wrapper.find('.el-pagination').exists()).toBe(false)
     })
-    wrapper.find('.el-pager').trigger('click')
-    expect(wrapper.vm.internalCurrentPage).toEqual(1)
   })
 
-  test('click li', () => {
-    const wrapper = mount(Pagination, {
-      props: {
-        total: 1000,
-      },
+  describe('test pageSize & currentPage reactive change', () => {
+    test(`test pageSize change`, async () => {
+      const pageSize = ref(10)
+      const wrapper = mount({
+        setup() {
+          return () => {
+            return h(Pagination, {
+              layout: 'pager',
+              total: 100,
+              pageSize: pageSize.value,
+            })
+          }
+        },
+      })
+      assertTotal(wrapper, 10/* 100/10 */)
```

Comment:
I think these comments are unnecessary since you can put a comment at the beginning of this test case:

` page = total / pageSize`